docs/modules/c-sharp/index.md

```diff
@@ -152,21 +152,46 @@ public enum Color
 
 SpacetimeDB has support for tagged enums which can be found in languages like Rust, but not C#.
 
-To bridge the gap, a special marker interface `SpacetimeDB.TaggedEnum` can be used on any `SpacetimeDB.Type`-marked `struct` or `class` to mark it as a SpacetimeDB tagged enum. It accepts a tuple of 2 or more named items and will generate methods to check which variant is currently active, as well as accessors for each variant.
+To bridge the gap, we allow to declare tagged enums by inheriting from a special marker `record SpacetimeDB.TaggedEnum<(...types and names of the variants as a tuple...)>`.
 
-It is expected that you will use the `Is*` methods to check which variant is active before accessing the corresponding field, as the accessor will throw an exception on a state mismatch.
+It will generate variants as subclasses of the marked type, so you can use regular C# pattern matching operators like `is` or `switch` to determine which variant a given tagged enum holds at any time.
+
+For unit variants (those without any data payload) you can use a built-in `SpacetimeDB.Unit` as the variant type.
```

Comment:
```suggestion
For unit variants (those without any data payload), declare `SpacetimeDB.Unit` as the variant type.
```

Reply:
I don't like "declare" in this context because it can be somewhat more ambiguous - people often think of it more as "declare your own type".

Reply:
I'll leave this one as-is.

Reply:
The wording as-is is awkward; "a built-in `SpacetimeDB.Unit`" doesn't scan right, and "you can" is too soft. This should be phrased in the imperative, not the passive, and the reference to `SpacetimeDB.Unit` should be a proper noun phrase.

Reply:
Well it's "you can" because users can certainly put their own empty struct if they want to instead. This one is provided for convenience.